opslevel/resources_opslevel_team_tag.go

```diff
@@ -1,159 +1,234 @@
+				Required:    true,
+			},
+			"team": schema.StringAttribute{
+				Description: "The id of the team that this will be added to.",
+				Optional:    true,
+			},
+			"team_alias": schema.StringAttribute{
+				Description: "The alias of the team that this will be added to.",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func (teamTagResource *TeamTagResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data TeamTagResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tagCreateInput := opslevel.TagCreateInput{
+		Type:  opslevel.RefOf(opslevel.TaggableResourceTeam),
+		Key:   data.Key.ValueString(),
+		Value: data.Value.ValueString(),
+	}
+
+	// use either the team ID or alias based on what is used in the config
+	var teamIdentifier string
+	if data.Team.ValueString() == "" && data.TeamAlias.ValueString() == "" {
+		resp.Diagnostics.AddError("config error", "require at least one of: 'team', 'team_alias'")
+		return
+	}
+	if data.Team.ValueString() != "" {
+		teamIdentifier = data.Team.ValueString()
+		tagCreateInput.Id = opslevel.NewID(teamIdentifier)
+	} else {
+		teamIdentifier = data.TeamAlias.ValueString()
+		tagCreateInput.Alias = &teamIdentifier
+	}
+
+	team, err := teamTagResource.client.CreateTag(tagCreateInput)
+	if err != nil || team == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to create team (%s) tag (with key '%s'), got error: %s", teamIdentifier, data.Key.ValueString(), err))
+		return
+	}
+
+	createdTeamTagResourceModel := NewTeamTagResourceModel(*team, teamIdentifier)
+	createdTeamTagResourceModel.LastUpdated = timeLastUpdated()
+	tflog.Trace(ctx, "created a team tag resource")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &createdTeamTagResourceModel)...)
+}
+
+func (teamTagResource *TeamTagResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data TeamTagResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// use either the team ID or alias based on what is used in the config
+	var teamIdentifier string
+	var team *opslevel.Team
+	var err error
+	if data.Team.ValueString() == "" && data.TeamAlias.ValueString() == "" {
+		resp.Diagnostics.AddError("config error", "require at least one of: 'team', 'team_alias'")
+		return
+	}
+	if data.Team.ValueString() != "" {
+		teamIdentifier = data.Team.ValueString()
+		team, err = teamTagResource.client.GetTeam(opslevel.ID(teamIdentifier))
+	} else {
+		teamIdentifier = data.TeamAlias.ValueString()
+		team, err = teamTagResource.client.GetTeamWithAlias(teamIdentifier)
+	}
+	if err != nil || team == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to read team (%s), got error: %s", teamIdentifier, err))
+		return
+	}
+	_, err = team.GetTags(teamTagResource.client, nil)
```

Comment:
I'm curious about this. `team.GetTags()` returns `TagConnection` with `Nodes` (of type `[]opslevel.Tag`). Is this the same `[]opslevel.Tag` that's being used in `for _, readTag := range team.Tags.Nodes {` a few lines down?
This doesn't necessarily need updating, I just find it surprising